backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -125,22 +126,21 @@ public CommCareSession(CommCareSession oldCommCareSession) {
      */
     private Vector<Entry> getEntriesForCommand(String commandId,
                                                OrderedHashtable<String, String> currentSessionData) {
+        Vector<Entry> entries = new Vector<>();
         for (Suite s : platform.getInstalledSuites()) {
-            for (Menu m : s.getMenus()) {
-                // We need to see if everything in this menu can be matched
-                if (commandId.equals(m.getId())) {
-                    return getEntriesFromMenu(m, currentSessionData);
+            List<Menu> menusWithId = s.getMenusWithId(commandId);
+            if (menusWithId != null) {
+                for (Menu menu : menusWithId) {
+                    entries.addAll(getEntriesFromMenu(menu, currentSessionData));
```

Comment:
Not sure, I wanted to get the big ones out of the way, but I'll reasses that issue soon. Might be the case that Kriti can use advanced modules instead.